crates/pop-drink/src/lib.rs

```diff
@@ -1,95 +1,150 @@
-            let error = deployment_result.unwrap().result.data;
-            return Err(E::decode(&mut &error[2..]).unwrap());
-        }
-        Ok(result.unwrap())
-    }
+/// Deploys a contract with a given constructo methodr, arguments, salt and initial value. In case
+/// of success, returns the address of the deployed contract.
+///
+/// # Generic Parameters:
+///
+/// - `S`: Sandbox defines the API of a sandboxed runtime.
+/// - `E`: Decodable error type returned if the contract deployment fails.
+///
+/// # Parameters:
+///
+/// - `session` - Wrapper around `Sandbox`` that provides a convenient API for interacting with
+///   multiple contracts.
+/// - `bundle` - A struct representing the result of parsing a `.contract` bundle file.
+/// - `method` - The name of the constructor method.
+/// - `input` - Arguments passed to the constructor method.
+/// - `salt` - Additional data used to influence the contract deployment address.
+/// - `init_value` - Initial funds allocated for the contract.
+pub fn deploy<S, E>(
+	session: &mut Session<S>,
+	bundle: ContractBundle,
+	method: &str,
+	input: Vec<String>,
+	salt: Vec<u8>,
+	init_value: Option<BalanceFor<S::Runtime>>,
+) -> Result<AccountIdFor<S::Runtime>, E>
+where
+	S: Sandbox,
+	S::Runtime: pallet_contracts::Config,
+	E: Decode,
+{
+	let result = session.deploy_bundle(bundle, method, &input, salt, init_value);
+	if result.is_err() {
+		let deployment_result = session.record().last_deploy_result().result.clone();
+		let error = deployment_result.unwrap().result.data;
+		return Err(E::decode(&mut &error[2..]).unwrap());
+	}
+	Ok(result.unwrap())
+}
 
-    /// Call a contract method and decode the returned data.
-    pub fn call<S, O, E>(
-        session: &mut Session<S>,
-        func_name: &str,
-        input: Vec<String>,
-        endowment: Option<BalanceFor<S::Runtime>>,
-    ) -> Result<O, E>
-    where
-        S: Sandbox,
-        S::Runtime: pallet_contracts::Config,
-        O: Decode,
-        E: Decode,
-    {
-        match session.call::<String, ()>(func_name, &input, endowment) {
-            // If the call is reverted, decode the error into `PSP22Error`.
-            Err(SessionError::CallReverted(error)) =>
-                Err(E::decode(&mut &error[2..])
-                    .unwrap_or_else(|_| panic!("Decoding failed"))),
-            // If the call is successful, decode the last returned value.
-            Ok(_) => Ok(session
-                .record()
-                .last_call_return_decoded::<O>()
-                .unwrap_or_else(|_| panic!("Expected a return value"))
-                .unwrap_or_else(|_| panic!("Decoding failed"))),
-            // Catch-all for unexpected results.
-            _ => panic!("Expected call to revert or return a value"),
-        }
-    }
+/// Call a contract method and decode the returned data.
+///
+/// # Generic Parameters:
+///
+/// - `S` - Sandbox defines the API of a sandboxed runtime.
```

Comment:
Reading this again this sentence doesn't make sense to me (so also the one above for deploy). We want the developers to understand these apis. A sandbox environment for the Pop runtime.